static/js/util/date_validation.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import R from 'ramda';
+import moment from 'moment';
+
+import { S, ifNil } from './sanctuary';
+const { Maybe, Nothing } = S;
+import { filterPositiveInt } from './util';
+import { YEAR_VALIDATION_CUTOFF } from '../constants';
+
+/**
+ * Removes non-numeric characters and truncates output string
+ */
+export const sanitizeNumberString = R.curry((length: number, input: string|number): string => {
+  if ( typeof input === 'string' ) {
+    let out = input.replace(/[^\d]+/g, '');
+    if ( out.match(/^0+/) ) {
+      if ( out.length <= length ) {
+        return out.slice(0, length);
+      } else {
+        return out.replace(/^0+/, "").slice(0, length);
+      }
+    } else {
+      return out.slice(0, length);
+    }
+  } else {
+    return String(input).slice(0, length);
```

Comment:
Can we move this to the beginning and do this even if the input is a string? It would reduce the number of branches which need testing